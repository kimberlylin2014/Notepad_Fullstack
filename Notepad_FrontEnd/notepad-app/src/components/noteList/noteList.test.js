import { shallow } from 'enzyme';
import React from 'react';
import NoteList from './noteList.component';

it('Testing Note List Component when user has more than one comments', () => {
    const mockUserPosts = [{
        created: "Sun 9/6/20 1:43am",
        id: 203,
        modified: true,
        post: "1"
    }]
    const mockUser ={id: 75, name: "kimberly", email: "kimberly@notepad.com", joined: "2020-08-31T03:19:15.374Z", comments: ["1"]} 
    const mockGetUserPostsStart = function getUserPosts() {}
    expect(shallow(<NoteList currentUser={mockUser} userPosts={mockUserPosts} getUserPostsStart={mockGetUserPostsStart}/>)).toMatchSnapshot();
})

it('Testing Note List Component when user has zero comments', () => {
    const mockUserPosts = []
    const mockUser ={id: 75, name: "kimberly", email: "kimberly@notepad.com", joined: "2020-08-31T03:19:15.374Z", comments: ["1"]} 
    const mockGetUserPostsStart = function getUserPosts() {}
    const wrapper = shallow(<NoteList currentUser={mockUser} userPosts={mockUserPosts} getUserPostsStart={mockGetUserPostsStart}/>);
    expect(wrapper.find('[id="list-row"]').text()).toEqual("");
})

it('Testing Note List Component when user has zero comments, header should be Add New Note', () => {
    const mockUserPosts = []
    const mockUser ={id: 75, name: "kimberly", email: "kimberly@notepad.com", joined: "2020-08-31T03:19:15.374Z", comments: ["1"]} 
    const mockGetUserPostsStart = function getUserPosts() {}
    const wrapper = shallow(<NoteList currentUser={mockUser} userPosts={mockUserPosts} getUserPostsStart={mockGetUserPostsStart}/>);
    expect(wrapper.find('[id="list-header"]').text()).toEqual("Add A Note!");
})

it('Testing Note List Component when user has zero comments, header should be {UserName}s Note', () => {
    const mockUserPosts = [{
        created: "Sun 9/6/20 1:43am",
        id: 203,
        modified: true,
        post: "1"
    }]
    const mockUser ={id: 75, name: "kimberly", email: "kimberly@notepad.com", joined: "2020-08-31T03:19:15.374Z", comments: ["1"]} 
    const userName = mockUser.name[0].toUpperCase() + mockUser.name.slice(1);
    const mockGetUserPostsStart = function getUserPosts() {}
    const wrapper = shallow(<NoteList currentUser={mockUser} userPosts={mockUserPosts} getUserPostsStart={mockGetUserPostsStart}/>);
    expect(wrapper.find('[id="list-header"]').text()).toEqual(`${userName}'s Notes`);
})


it('Testing Note List Component when current user has zero comments', () => {
    const mockUserPosts = []
    const mockUser ={id: 75, name: "kimberly", email: "kimberly@notepad.com", joined: "2020-08-31T03:19:15.374Z", comments: []} 
    const mockGetUserPostsStart = function getUserPosts() {}
    const wrapper = shallow(<NoteList currentUser={mockUser} userPosts={mockUserPosts} getUserPostsStart={mockGetUserPostsStart}/>);
    expect(wrapper.find('[id="list-row"]').text()).toEqual("");
})

