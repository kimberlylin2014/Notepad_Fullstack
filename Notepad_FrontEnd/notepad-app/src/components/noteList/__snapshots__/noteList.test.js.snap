// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Note List Component when user has more than one comments 1`] = `
<div
  className="NoteList"
>
  <h2
    id="list-header"
  >
    Kimberly's Notes
  </h2>
  <div
    className="container"
  >
    <div
      className="row"
      id="list-row"
    >
      <div
        className="col-md-4 col-6 eachRow"
        key="203"
      >
        <NoteCard
          currentUser={
            Object {
              "comments": Array [
                "1",
              ],
              "email": "kimberly@notepad.com",
              "id": 75,
              "joined": "2020-08-31T03:19:15.374Z",
              "name": "kimberly",
            }
          }
          getUserPostsStart={[Function]}
          postData={
            Object {
              "created": "Sun 9/6/20 1:43am",
              "id": 203,
              "modified": true,
              "post": "1",
            }
          }
          userPosts={
            Array [
              Object {
                "created": "Sun 9/6/20 1:43am",
                "id": 203,
                "modified": true,
                "post": "1",
              },
            ]
          }
        />
      </div>
    </div>
  </div>
</div>
`;
